test/factory/dart_function_factory_test.dart

```diff
@@ -0,0 +1,90 @@
+// ignore_for_file: deprecated_member_use_from_same_package
+// ignore_for_file: invalid_use_of_protected_member
+@TestOn('browser')
+import 'dart:js';
+
+import 'package:test/test.dart';
+
+import 'package:react/react.dart' as react;
+import 'package:react/react_client.dart';
+import 'package:react/react_client/react_interop.dart';
+
+import '../util.dart';
+import 'common_factory_tests.dart';
+
+main() {
+  setClientConfiguration();
+
+  group('ReactDartFunctionComponentFactoryProxy', () {
+    test('has a type corresponding to the backing JS Function', () {
+      expect(FunctionFoo.type, equals(FunctionFoo.reactFunction));
+    });
+
+    group('Function Component -', () {
+      group('- common factory behavior -', () {
+        commonFactoryTests(FunctionFoo, isFunctionComponent: true);
+      });
+    });
+
+    group('utils', () {
```

Comment:
These utils tests (minus the addition of the function component cases) are duplicated from another file. Could we split these utils out into a separate file? Sorry I didn't do that in the first place.